data/src/main/java/org/akvo/flow/data/datasource/preferences/SharedPreferencesDataSource.java

```diff
@@ -68,14 +70,35 @@ public SharedPreferencesDataSource(SharedPreferences prefs) {
         return Observable.just(getString(KEY_DEVICE_IDENTIFIER, DEFAULT_VALUE_DEVICE_IDENTIFIER));
     }
 
-    private int getInt(String key, int defaultValue) {
-        return preferences.getInt(key, defaultValue);
+    public Observable<Long> getSelectedSurvey() {
```

Comment:
it is the selected surveyId. We keep it to store the current state but also when the app restarts we have to display the selected survey. When user selects another survey, we store it in shared preferences. The current ui does not need it but since we use the current survey to highlight the selected survey in the Navigation drawer, but also in SurveyActivity to display survey name if any and inside the datapoint list to show the right error message (if there is a survey selected or if none is selected)